sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -324,12 +324,13 @@ private void populateDisplayData(DisplayData.Builder builder) {
   }
 
   /**
-   * Marker interface used when the original {@link PipelineOptions} interface is not available at
+   * Marker interface used when the original {@link PipelineOptions} interface is not known at
    * runtime. This can occur when {@link PipelineOptions} are deserialized from JSON or specified
    * on the command line.
    *
-   * To ensure {@link PipelineOptions} type information is available at runtime, register known
-   * interfaces via {@link PipelineOptionsFactory#register(Class)}.
+   * <p>Pipeline authors can ensure {@link PipelineOptions} type information is available at
```

Comment:
I would suggest users to look at the Registration section on PipelineOptions. The auto registration using service loader is preferred over static registration.